web/src/compose_ui.js

```diff
@@ -362,6 +363,26 @@ export function handle_keyup(_event, $textarea) {
     rtl.set_rtl_class_for_textarea($textarea);
 }
 
+export function cursor_inside_code_block($textarea) {
+    // Returns whether the cursor is at a point that would be inside
+    // a code block on rendering the textarea content as markdown.
+    const cursor_position = $textarea.caret();
+    const current_content = $textarea.val();
+
+    let unique_insert = "UNIQUEINSERT:" + Math.random();
+    while (current_content.includes(unique_insert)) {
+        unique_insert = "UNIQUEINSERT:" + Math.random();
```

Comment:
I made these strings a little more greppable in case one shows up somewhere else.